.travis.yml

```diff
@@ -31,16 +30,19 @@ install:
                :prompt nil)'
 
 script:
-  - ros -e '(ql:quickload :emotiq-test) 
-            (when (asdf:test-system :emotiq)
-               (uiop:quit 0))'
-  - ros -e '(ql:quickload :cosi-test)
-            (when (asdf:test-system :cosi)
-               (uiop:quit 0))'
-  # Speculatively hook into Coveralls <https://coveralls.io/>
-  # Coveralls invocation (currently unused)
-  #                 
-  # - ros -s prove -s cl-coveralls
-  #       -e '(or (coveralls:with-coveralls (:exclude (list "t"))
-  #                 (prove:run :emotiq-prove))
-  #               (uiop:quit -1))'
+  - ros -e "(loop
+              :for i :upfrom 0
+              :for 
+                  (system dependencies) 
+              :in 
+                  '((:emotiq :emotiq-test)
+                    (:cosi   :cosi-test))
+              :doing 
+                 (ql:quickload dependencies)
+              :unless 
+                 (asdf:test-system system)
+              :return 
+                 (uiop:quit (ash -1 i))
+              :finally 
+                 (uiop:quit 0))"
```

Comment:
;; success

Reply:
We could some day do a fancier status code with both a run number (the 'i') and a type-of-problem report. I wrote a prototype Lisp function for this in a research branch for this. Check it out: https://github.com/emotiq/research/blob/status-code-ci/status-code-ci.lisp
But I'd suggest deferring on that for now and go with the simpler approach above.

Reply:
@markhdavid Total bikeshed on the return value, but nice code!